src/main/java/com/github/checkstyle/regression/Main.java

```diff
@@ -34,6 +74,168 @@ private Main() {
      * @throws Exception execute failure
      */
     public static void main(String[] args) throws Exception {
-        // empty for now
+        final Options options = createOptions();
+        final CommandLineParser parser = new DefaultParser();
+        final HelpFormatter formatter = new HelpFormatter();
+        CommandLine cmd = null;
+
+        try {
+            cmd = parser.parse(options, args);
+        }
+        catch (ParseException ex) {
+            System.err.println(ex.getMessage());
+            formatter.printHelp(null, options);
```

Comment:
Fixed. `cmdLineSyntax` is not allowed to be null or empty. Now I pass a `regression-tool` to it. (the argument is shown after `usage:`)
Acts like:
```bash
$ java -jar target/regression-tool-1.0-SNAPSHOT-all.jar
Missing required options: r, p
usage: regression-tool
 -p,--patchBranch <arg>            the name of the PR branch
 -r,--localGitRepo <arg>           the path of the checkstyle repository
    --stopAfterGenerateConfig      indicates that regression tool would
                                   stop after generating config
 -t,--checkstyleTesterPath <arg>   the path of the checkstyle-tester
                                   directory
```